Error during build in Ubuntu 17.10

Hey there!

I was compiling the openMVS in Ubuntu 17.10. I follow all steps of build tutorial,but unfortunatelly I had this error:

make -j2

```
[  1%] Building CXX precompiled header libs/Common/cotire/Common_CXX_prefix.hxx.gch
In file included from /usr/local/include/eigen3/Eigen/Core:326:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:122,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.cxx:4,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx:4:
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h: In member function ‘void Eigen::internal::gebp_kernel<LhsScalar, RhsScalar, Index, mr, nr, ConjugateLhs, ConjugateRhs>::operator()(Eigen::internal::gebp_kernel<LhsScalar, RhsScalar, Index, mr, nr, ConjugateLhs, ConjugateRhs>::ResScalar*, Index, const LhsScalar*, const RhsScalar*, Index, Index, Index, Eigen::internal::gebp_kernel<LhsScalar, RhsScalar, Index, mr, nr, ConjugateLhs, ConjugateRhs>::ResScalar, Index, Index, Index, Index, RhsScalar*)’:
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:574:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(nr==4) traits.initAcc(C3);
         ^~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:575:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   traits.initAcc(C4);
                   ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:946:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(nr==4) R3 = ploadu<ResPacket>(r3);
         ^~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:948:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   traits.acc(C0, alphav, R0);
                   ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:951:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(nr==4) traits.acc(C3, alphav, R3);
         ^~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:953:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   pstoreu(r0, R0);
                   ^~~~~~~
In file included from /usr/local/include/eigen3/Eigen/SparseLU:29:0,
                 from /usr/local/include/eigen3/Eigen/Sparse:22,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:124,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.cxx:4,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx:4:
/usr/local/include/eigen3/Eigen/src/SparseLU/SparseLU_gemm_kernel.h: In function ‘void Eigen::internal::sparselu_gemm(Index, Index, Index, const Scalar*, Index, const Scalar*, Index, Scalar*, Index)’:
/usr/local/include/eigen3/Eigen/src/SparseLU/SparseLU_gemm_kernel.h:78:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(RK==4) b30 = pset1<Packet>(Bc0[3]);
         ^~
/usr/local/include/eigen3/Eigen/src/SparseLU/SparseLU_gemm_kernel.h:79:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   b01 = pset1<Packet>(Bc1[0]);
                   ^~~
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:369:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.cxx:4,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx:4:
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h: In static member function ‘static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(const SEACAVE::String&)’:
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:246:87: error: no matching function for call to ‘SEACAVE::cHashTable<Type>::HashKey(const char*, std::__cxx11::basic_string<char>::size_type)’
  static inline Key HashKey(const String& str) { return HashKey(str.c_str(), str.size()); }
                                                                                       ^
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:244:20: note: candidate: static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(const uint8_t*, UINT) <near match>
  static inline Key HashKey(const uint8_t* data, UINT size) { return HashKeyFNV(data, size); }
                    ^~~~~~~
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:244:20: note:   conversion of argument 1 would be ill-formed:
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:245:20: note: candidate: static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(LPCTSTR)
  static inline Key HashKey(LPCTSTR sz) { return HashKey((const uint8_t*)sz, _tcslen(sz)); }
                    ^~~~~~~
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:245:20: note:   candidate expects 1 argument, 2 provided
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:246:20: note: candidate: static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(const SEACAVE::String&)
  static inline Key HashKey(const String& str) { return HashKey(str.c_str(), str.size()); }
                    ^~~~~~~
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:246:20: note:   candidate expects 1 argument, 2 provided
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:377:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.cxx:4,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx:4:
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h: In static member function ‘static short int SEACAVE::hfloat::fromFloat(float)’:
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h:62:29: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   const int& fltInt32((int&)fv);
                             ^~
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h: In static member function ‘static float SEACAVE::hfloat::toFloat(short int)’:
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h:70:18: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   return (float&)fltInt32;
                  ^~~~~~~~
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:425:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.cxx:4,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx:4:
/home/cogitas3d/Programs/openMVS/libs/Common/SML.h: In destructor ‘SEACAVE::SMLVALUE_TYPE::~SMLVALUE_TYPE()’:
/home/cogitas3d/Programs/openMVS/libs/Common/SML.h:29:28: warning: deleting ‘void*’ is undefined [-Wdelete-incomplete]
  ~SMLVALUE_TYPE() { delete data; }
                            ^~~~
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176:0,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.cxx:4,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx:4:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h: In function ‘float SEACAVE::cbrt5(float)’:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:735:8: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  (int&)f = (((int&)f-(127<<23))/3+(127<<23));
        ^
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:735:20: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  (int&)f = (((int&)f-(127<<23))/3+(127<<23));
                    ^
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h: In member function ‘int SEACAVE::Float::Floor2Int() const’:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:1176:94: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  inline int  Floor2Int() const    { const double d((double)v.f + FLOOR_FIX); return (*(int*)&d) >> 16; }
                                                                                              ^
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h: In member function ‘int SEACAVE::Float::Ceil2Int() const’:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:1179:99: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  inline int  Ceil2Int() const    { const double d((double)v.f + 1.0 + FLOOR_FIX); return (*(int*)&d) >> 16; }
                                                                                                   ^
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:2730:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.cxx:4,
                 from /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx:4:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.inl: In member function ‘void SEACAVE::TImage<TYPE>::toGray(SEACAVE::TImage<T>&, int, bool) const’:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.inl:2065:33: error: cannot call member function ‘int cv::Mat::channels() const’ without object
  const int scn(cv::Mat::channels());
                ~~~~~~~~~~~~~~~~~^~
CMake Error at /home/cogitas3d/Programs/openMVS/build/Cotire.cmake:1826 (message):
  cotire: error 1 precompiling
  /home/cogitas3d/Programs/openMVS_build/libs/Common/cotire/Common_CXX_prefix.hxx.
Call Stack (most recent call first):
  /home/cogitas3d/Programs/openMVS/build/Cotire.cmake:3421 (cotire_precompile_prefix_header)


libs/Common/CMakeFiles/Common.dir/build.make:62: recipe for target 'libs/Common/cotire/Common_CXX_prefix.hxx.gch' failed
make[2]: *** [libs/Common/cotire/Common_CXX_prefix.hxx.gch] Error 1
CMakeFiles/Makefile2:179: recipe for target 'libs/Common/CMakeFiles/Common.dir/all' failed
make[1]: *** [libs/Common/CMakeFiles/Common.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

I did it on Types.inl

Run again and:

```
[  1%] Building CXX object libs/Common/CMakeFiles/Common.dir/Common.cpp.o
In file included from /usr/local/include/eigen3/Eigen/Core:326:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:122,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.cpp:12:
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h: In member function ‘void Eigen::internal::gebp_kernel<LhsScalar, RhsScalar, Index, mr, nr, ConjugateLhs, ConjugateRhs>::operator()(Eigen::internal::gebp_kernel<LhsScalar, RhsScalar, Index, mr, nr, ConjugateLhs, ConjugateRhs>::ResScalar*, Index, const LhsScalar*, const RhsScalar*, Index, Index, Index, Eigen::internal::gebp_kernel<LhsScalar, RhsScalar, Index, mr, nr, ConjugateLhs, ConjugateRhs>::ResScalar, Index, Index, Index, Index, RhsScalar*)’:
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:574:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(nr==4) traits.initAcc(C3);
         ^~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:575:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   traits.initAcc(C4);
                   ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:946:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(nr==4) R3 = ploadu<ResPacket>(r3);
         ^~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:948:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   traits.acc(C0, alphav, R0);
                   ^~~~~~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:951:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(nr==4) traits.acc(C3, alphav, R3);
         ^~
/usr/local/include/eigen3/Eigen/src/Core/products/GeneralBlockPanelKernel.h:953:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   pstoreu(r0, R0);
                   ^~~~~~~
In file included from /usr/local/include/eigen3/Eigen/SparseLU:29:0,
                 from /usr/local/include/eigen3/Eigen/Sparse:22,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:124,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.cpp:12:
/usr/local/include/eigen3/Eigen/src/SparseLU/SparseLU_gemm_kernel.h: In function ‘void Eigen::internal::sparselu_gemm(Index, Index, Index, const Scalar*, Index, const Scalar*, Index, Scalar*, Index)’:
/usr/local/include/eigen3/Eigen/src/SparseLU/SparseLU_gemm_kernel.h:78:9: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
         if(RK==4) b30 = pset1<Packet>(Bc0[3]);
         ^~
/usr/local/include/eigen3/Eigen/src/SparseLU/SparseLU_gemm_kernel.h:79:19: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
                   b01 = pset1<Packet>(Bc1[0]);
                   ^~~
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:369:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.cpp:12:
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h: In static member function ‘static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(const SEACAVE::String&)’:
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:246:87: error: no matching function for call to ‘SEACAVE::cHashTable<Type>::HashKey(const char*, std::__cxx11::basic_string<char>::size_type)’
 ine Key HashKey(const String& str) { return HashKey(str.c_str(), str.size()); }
                                                                            ^
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:244:20: note: candidate: static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(const uint8_t*, UINT) <near match>
  static inline Key HashKey(const uint8_t* data, UINT size) { return HashKeyFNV(data, size); }
                    ^~~~~~~
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:244:20: note:   conversion of argument 1 would be ill-formed:
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:245:20: note: candidate: static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(LPCTSTR)
  static inline Key HashKey(LPCTSTR sz) { return HashKey((const uint8_t*)sz, _tcslen(sz)); }
                    ^~~~~~~
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:245:20: note:   candidate expects 1 argument, 2 provided
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:246:20: note: candidate: static SEACAVE::cHashTable<Type>::Key SEACAVE::cHashTable<Type>::HashKey(const SEACAVE::String&)
  static inline Key HashKey(const String& str) { return HashKey(str.c_str(), str.size()); }
                    ^~~~~~~
/home/cogitas3d/Programs/openMVS/libs/Common/Hash.h:246:20: note:   candidate expects 1 argument, 2 provided
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:377:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.cpp:12:
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h: In static member function ‘static short int SEACAVE::hfloat::fromFloat(float)’:
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h:62:29: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   const int& fltInt32((int&)fv);
                             ^~
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h: In static member function ‘static float SEACAVE::hfloat::toFloat(short int)’:
/home/cogitas3d/Programs/openMVS/libs/Common/HalfFloat.h:70:18: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   return (float&)fltInt32;
                  ^~~~~~~~
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Types.h:425:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.cpp:12:
/home/cogitas3d/Programs/openMVS/libs/Common/SML.h: In destructor ‘SEACAVE::SMLVALUE_TYPE::~SMLVALUE_TYPE()’:
/home/cogitas3d/Programs/openMVS/libs/Common/SML.h:29:28: warning: deleting ‘void*’ is undefined [-Wdelete-incomplete]
  ~SMLVALUE_TYPE() { delete data; }
                            ^~~~
In file included from /home/cogitas3d/Programs/openMVS/libs/Common/Common.h:176:0,
                 from /home/cogitas3d/Programs/openMVS/libs/Common/Common.cpp:12:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h: In function ‘float SEACAVE::cbrt5(float)’:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:735:8: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  (int&)f = (((int&)f-(127<<23))/3+(127<<23));
        ^
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:735:20: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  (int&)f = (((int&)f-(127<<23))/3+(127<<23));
                    ^
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h: In member function ‘int SEACAVE::Float::Floor2Int() const’:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:1176:94: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
 const    { const double d((double)v.f + FLOOR_FIX); return (*(int*)&d) >> 16; }
                                                                     ^
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h: In member function ‘int SEACAVE::Float::Ceil2Int() const’:
/home/cogitas3d/Programs/openMVS/libs/Common/Types.h:1179:99: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
    { const double d((double)v.f + 1.0 + FLOOR_FIX); return (*(int*)&d) >> 16; }
                                                                     ^
libs/Common/CMakeFiles/Common.dir/build.make:62: recipe for target 'libs/Common/CMakeFiles/Common.dir/Common.cpp.o' failed
make[2]: *** [libs/Common/CMakeFiles/Common.dir/Common.cpp.o] Error 1
CMakeFiles/Makefile2:144: recipe for target 'libs/Common/CMakeFiles/Common.dir/all' failed
make[1]: *** [libs/Common/CMakeFiles/Common.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/libs/Common/Hash.h
+++ b/libs/Common/Hash.h
@@ -243,7 +243,7 @@ public:
 	}
 	static inline Key HashKey(const uint8_t* data, UINT size) { return HashKeyFNV(data, size); }
 	static inline Key HashKey(LPCTSTR sz) { return HashKey((const uint8_t*)sz, _tcslen(sz)); }
-	static inline Key HashKey(const String& str) { return HashKey(str.c_str(), str.size()); }
+	static inline Key HashKey(const String& str) { return HashKey((const uint8_t*)str.c_str(), (UINT)str.size()); }
 
 	// Convenience functions
 	inline		 Type*	Find		(LPCTSTR		key)			{ return  Find  ( HashKey(key) );			}

--- a/libs/Common/Hash.h
+++ b/libs/Common/Hash.h
@@ -242,7 +242,7 @@ public:
 		return key;
 	}
 	static inline Key HashKey(const uint8_t* data, UINT size) { return HashKeyFNV(data, size); }
-	static inline Key HashKey(LPCTSTR sz) { return HashKey((const uint8_t*)sz, _tcslen(sz)); }
+	static inline Key HashKey(LPCTSTR sz) { return HashKey((const uint8_t*)sz, (UINT)_tcslen(sz)); }
 	static inline Key HashKey(const String& str) { return HashKey((const uint8_t*)str.c_str(), (UINT)str.size()); }
 
 	// Convenience functions

--- a/libs/Common/Types.inl
+++ b/libs/Common/Types.inl
@@ -2060,7 +2060,7 @@ void TImage<TYPE>::toGray(TImage<T>& out, int code, bool bNormalize) const
 	const T &cb(coeffs[0]), &cg(coeffs[1]), &cr(coeffs[2]);
 	if (out.rows!=rows || out.cols!=cols)
 		out.create(rows, cols);
-	ASSERT(cv::Mat::isContinuous());
+	ASSERT(this->isContinuous());
 	ASSERT(out.cv::Mat::isContinuous());
 	const int scn(this->channels());
 	T* dst = out.cv::Mat::template ptr<T>();
@@ -2274,7 +2274,7 @@ inline void _ProcessScanLine(int y, const TPoint3<T>& pa, const TPoint3<T>& pb, 
 	}
 }
 // Raster the given triangle and output the position and depth of each pixel of the triangle;
-// based on "Learning how to write a 3D software engine – Rasterization & Z-Buffering" by Nick (David Rousset)
+// based on "Learning how to write a 3D software engine Â– Rasterization & Z-Buffering" by Nick (David Rousset)
 // http://blogs.msdn.com/b/davrous/archive/2013/06/21/tutorial-part-4-learning-how-to-write-a-3d-software-engine-in-c-ts-or-js-rasterization-amp-z-buffering.aspx
 template <typename TYPE>
 template <typename T, typename PARSER>

--- a/libs/Common/Types.inl
+++ b/libs/Common/Types.inl
@@ -2062,7 +2062,7 @@ void TImage<TYPE>::toGray(TImage<T>& out, int code, bool bNormalize) const
 		out.create(rows, cols);
 	ASSERT(cv::Mat::isContinuous());
 	ASSERT(out.cv::Mat::isContinuous());
-	const int scn(cv::Mat::channels());
+	const int scn(this->channels());
 	T* dst = out.cv::Mat::template ptr<T>();
 	T* const dstEnd = dst + out.area();
 	typedef typename cv::DataType<TYPE>::channel_type ST;

